tests/test_artifacts.py

```diff
@@ -501,3 +502,26 @@ def test_update_property_fail_bad_value():
             {BAD_PROPERTY_NAME: [BAD_PROPERTY_VALUE]},
         )
         assert update_properties_response is None
```

Comment:
Can you please add a new test `test_deploy_artifact_with_checksum_success` where you set the headers on the PUT request?